optuna/study/_multi_objective.py

```diff
@@ -69,6 +74,94 @@ def _get_pareto_front_trials(study: "optuna.study.Study") -> List[FrozenTrial]:
     return _get_pareto_front_trials_by_trials(study.trials, study.directions)
 
 
+def _fast_non_dominated_sort(
+    objective_values: np.ndarray,
+    *,
+    penalty: np.ndarray | None = None,
+    n_below: int | None = None,
+) -> np.ndarray:
```

Comment:
It would be great to leave the shape and its meaning of the returned ndarray. Could you add the explanation?